code/datums/expansions/multitool/machinery/mineral.dm

```diff
@@ -0,0 +1,36 @@
+/datum/expansion/multitool/mineral/get_interact_window(var/obj/item/device/multitool/M, var/mob/user)
+	. += buffer(M)
+	. += "<HR><b>Connected Machine:</b> "
+	var/obj/machinery/mineralconsole/MC = holder
+	if(MC.machine)
+		. += "[MC.machine.name]<br>"
+
+/datum/expansion/multitool/mineral/receive_buffer(var/obj/item/device/multitool/M, var/atom/buffer, var/mob/user)
+	var/obj/machinery/mineral/MM = buffer
+	var/obj/machinery/mineralconsole/C = holder
+
+	if(!istype(M))
+		to_chat(user, span("warning", "No valid connection data in \the [M] buffer."))
+		return MT_NOACTION
+
+	if(C.machine)
+		to_chat(user, span("notice", "You disconnect [C.machine] from [C]."))
+		C.Unlink()
+		return MT_REFRESH
+	if(MM.console)
+		to_chat(user, span("notice", "There is already a console connected to that machine!"))
+		return MT_NOACTION
+
+	var/is_connected = C.machine
+	if(!is_connected)
```

Comment:
Line 16 is `if (C.machine) return;`

So checking the inverse of this afterwards is a tautology.